tests/testthat/test-class-Megastudy.R

```diff
@@ -544,4 +548,42 @@ test_that("imputeZeroes method is sane", {
   ))
 
   expect_error(getDTWithImputedZeroes(m, variables))
+
+  # variable collection exists in plot
+  pathogenVariableCollectionVocabs <- StudySpecificVocabulariesByVariable(S4Vectors::SimpleList(StudySpecificVocabulary(studyIdColumnName='study.id', study='a', variableSpec=VariableSpec(entityId='assay',variableId='pathogen_presence_variable_collection'), vocabulary=c('Yes','No')),
+                                                                                                StudySpecificVocabulary(studyIdColumnName='study.id', study='b', variableSpec=VariableSpec(entityId='assay',variableId='pathogen_presence_variable_collection'), vocabulary=c('Yes','No'))))
+
+
+  m <- Megastudy(data=megastudyDT,
+                 ancestorIdColumns=c('study.id', 'collection.id', 'sample.id', 'assay.id'),
+                 studySpecificVocabularies=StudySpecificVocabulariesByVariableList(S4Vectors::SimpleList(pathogenVariableCollectionVocabs)))
+
+  variables <- new("VariableMetadataList", SimpleList(
+    new("VariableMetadata",
+      variableClass = new("VariableClass", value = 'native'),
+      variableSpec = new("VariableSpec", variableId = 'pathogen_presence_variable_collection', entityId = 'assay'),
+      plotReference = new("PlotReference", value = 'overlay'),
+      dataType = new("DataType", value = 'STRING'),
+      dataShape = new("DataShape", value = 'CATEGORICAL'),
+      isCollection = TRUE,
+      weightingVariableSpec = VariableSpec(variableId='specimen_count',entityId='sample'),
+      hasStudyDependentVocabulary = TRUE,
+      members = VariableSpecList(S4Vectors::SimpleList(VariableSpec(variableId='pathogen_presence', entityId='assay'),
+                                                       VariableSpec(variableId='pathogen2_presence', entityId='assay'),
+                                                       VariableSpec(variableId='pathogen3_presence', entityId='assay')))
```

Comment:
i think i need a test where i have a collection, but the OTHER var is the one w the special vocab